src/ros/ROSInterface.cpp

```diff
@@ -0,0 +1,268 @@
+    iai_knowledge_msgs::askallResult result;
+    while(true) {
+        auto nextResult = resultQueue->pop_front();
+
+        if(AnswerStream::isEOS(nextResult)) {
+            break;
+        }
+        else {
+            if (nextResult->substitution()->empty()) {
+                numSolutions_ = 1;
+                break;
+            } else {
+                // Push one answer
+                iai_knowledge_msgs::GraphAnswer answer = createGraphAnswer(nextResult);
+                result.answer.push_back(answer);
+                numSolutions_ += 1;
+                // publish feedback
+                iai_knowledge_msgs::askallFeedback feedback;
+                feedback.numberOfSolutions = numSolutions_;
+                askall_action_server_.publishFeedback(feedback);
+            }
+        }
+    }
+
+    if(numSolutions_ == 0) {
+        result.status = iai_knowledge_msgs::askallResult::FALSE;
+    } else {
+        result.status = iai_knowledge_msgs::askallResult::TRUE;
+    }
+    askall_action_server_.setSucceeded(result);
+}
+
+void ROSInterface::executeAskOneCB(const iai_knowledge_msgs::askoneGoalConstPtr& goal)
+{
+    // Implement your action here
+    FormulaPtr phi(QueryParser::parse(goal->query.queryString));
+
+    FormulaPtr mPhi = applyModality(goal->query, phi);
+    auto query = std::make_shared<ModalQuery>(mPhi, QUERY_FLAG_ONE_SOLUTION);
+
+    auto resultStream = kb_.submitQuery(query->formula(), QUERY_FLAG_ONE_SOLUTION);
+    auto resultQueue = resultStream->createQueue();
+
+    iai_knowledge_msgs::askoneResult result;
+    auto nextResult = resultQueue->pop_front();
+
+    if(AnswerStream::isEOS(nextResult)) {
+        result.status = iai_knowledge_msgs::askoneResult::FALSE;
+    } else {
+        result.status = iai_knowledge_msgs::askoneResult::TRUE;
+        if (!nextResult->substitution()->empty()) {
+            iai_knowledge_msgs::GraphAnswer answer = createGraphAnswer(nextResult);
+            result.answer = answer;
+        }
+    }
+
+    iai_knowledge_msgs::askoneFeedback  feedback;
+    feedback.finished = true;
+    askone_action_server_.publishFeedback(feedback);
+    askone_action_server_.setSucceeded(result);
+}
+
+//void ROSInterface::executeAskIterativeCB(const iai_knowledge_msgs::askiterativeGoalConstPtr& goal)
+//{
+//    // Implement your action here
+//
+//    // By default, saying the action was successful
+//    askiterative_action_server_.setSucceeded();
+//}
+
+boost::property_tree::ptree loadSetting() {
+    // Check for settings file
+    std::string config_path = "default.json";
+    if(std::getenv("KNOWROB_SETTINGS")) {
+        config_path = std::getenv("KNOWROB_SETTINGS");
+    }
+
+    // read the settings
+    boost::property_tree::ptree config;
+    boost::property_tree::read_json(
+            config_path,
+            config);
+
+    return config;
+
+}
+
+int main(int argc, char **argv) {
+	InitKnowledgeBase(argc, argv);
+
+    // Load settings files
+    loadSetting();
```

Comment:
seems nothing is done with settings here! The `KnowledgeBase` should be initialized with the settings

Reply:
That call seems to be useless, so I removed it. loadSetting is used to load the knowledgegraph by the call in 261